tools/tool-env/profile.js

```diff
@@ -280,7 +280,11 @@ function Profile(bucketName, f) {
     var start = process.hrtime();
     var err = null;
     try {
-      return f.apply(this, arguments);
+      const result = f.apply(this, arguments);
+      if (result && typeof result.then === "function") {
+        return Promise.await(result);
```

Comment:
Since the `Profile` abstraction assumes each chunk of hierarchical work will run to completion, we have to keep pretending async work is synchronous. If anyone has a better way to instrument asynchronous code for performance measurement, I'd love to hear about it!

Reply:
Unfortunately, this `Promise.await` behavior does not apply when `METEOR_PROFILE` is disabled, because then the `Profile` function just returns the original function without wrapping it. That's a huge difference in behavior that I'm not willing to accept, even though we can sort of work around it by `await`ing the result of any `Promise`-returning `Profile`-wrapped function. I'll have to come up with another approach.